src/Main.ts

```diff
@@ -1135,6 +1141,23 @@ export class Main {
         log.info("Bridge killed");
     }
 
+    private async onRoomUpgrade(oldRoomId: string, newRoomId: string) {
+        log.info(`Room has been upgraded from ${oldRoomId} to ${newRoomId}`);
+        const bridgedroom = this.rooms.getByMatrixRoomId(oldRoomId);
+        const adminRoomUser = await this.datastore.getUserForAdminRoom(oldRoomId);
+        if (bridgedroom) {
+            log.info("Migrating channel");
+            this.rooms.removeRoom(bridgedroom);
+            bridgedroom.migrateToNewRoomId(newRoomId);
+            this.rooms.upsertRoom(bridgedroom);
+            await this.datastore.upsertRoom(bridgedroom);
+        } else if (adminRoomUser) {
```

Comment:
If the admin room cannot be a bridged room, feel free to merge this.

Reply:
It really shouldn't be, no :p